python_modules/dagster/dagster/_core/definitions/inference.py

```diff
@@ -101,7 +111,8 @@ def _infer_inputs_from_params(
 def infer_input_props(fn: Callable, context_arg_provided: bool) -> Sequence[InferredInputProps]:
     sig = signature(fn)
     params = list(sig.parameters.values())
+    type_hints = get_type_hints(fn)
     descriptions = _infer_input_description_from_docstring(fn)
     params_to_infer = params[1:] if context_arg_provided else params
-    defs = _infer_inputs_from_params(params_to_infer, descriptions=descriptions)
+    defs = _infer_inputs_from_params(params_to_infer, type_hints, descriptions=descriptions)
```

Comment:
unfortunate how this similar but slightly different information flows through from inspect and typing

Reply:
indeed, the python APIs around type annotations are a bit of a mess